core/src/test/scala/com/github/johnynek/paiges/PaigesTest.scala

```diff
@@ -127,6 +121,86 @@ the spaces""")
       else succeed
     }
   }
+  // test("renders are constant after maxWidth") {
+  //   forAll { (d: Doc, ws: List[Int]) =>
+  //     val m = Doc.maxWidth(d)
+  //     val maxR = d.render(m)
+  //     val goodW = ws.map { w => (m + w) max m }
+  //     assert(goodW.forall { w =>
+  //       val wrender = d.render(w)
+  //       //if (wrender != maxR) { println(s"$wrender != $maxR\nwidth=$w\nmax=$m"); false }
+  //       (wrender == maxR)
+  //     })
+  //   }
+  // }
+  // test("either all widths render the same or max-1 renders differently") {
+  //   forAll { (d: Doc) =>
+  //     val m = Doc.maxWidth(d)
+  //     if (m == 0) succeed
+  //     else {
+  //       val md = d.render(m)
+  //       val allSame = !((0 to m).sliding(2).exists { v => d.render(v(0)) != d.render(v(1)) })
+  //       assert(allSame || (d.render(m - 1) != md))
+  //     }
+  //   }
+  // }
+  test("if we always render the same, we compare the same") {
+    forAll { (a: Doc, b: Doc) =>
+      if (a.compare(b) != 0) succeed
+      else {
+        //val maxR = Doc.maxWidth(a) max Doc.maxWidth(b)
+        val maxR = 1000
```

Comment:
if we had a true maximum, we could reverse this, but right now it is weaker (if we are equal we render the same).